cwl/src/test/scala/wdl4s/cwl/ThreeStepSpec.scala

```diff
@@ -1,51 +1,173 @@
+    requirements = inlineJScriptRequirements
+  )
+
+  val patternCgrepWorkFlowStepInput = WorkflowStepInput(id = "pattern", source = Some(Coproduct("#pattern")))
+
+  val fileCgrepWorkflowStepInput = WorkflowStepInput(id = "file", source = Some(Coproduct("ps/ps-stdOut")))
+
+  val grepWfStep  = WorkflowStep(
+    id = Some("cgrep"),
+    in = Coproduct(Array(patternCgrepWorkFlowStepInput, fileCgrepWorkflowStepInput)),
+    out = Coproduct[Outputs](Array(WorkflowStepOutput("cgrep-stdOut"))),
+    run = Coproduct[Run](cgrepClt))
+
+
+  /**
+    * WC
+    */
+  val wcFileCommandInput = CommandInputParameter(
+    id = Some("file"),
+    `type` = Some(Coproduct(CwlType.File)))
+
+  val wcArgs =
+    Array(
+      clb("cat"),
+      clb("$(inputs.file)"),
+      clb("|"),
+      clb("wc"),
+      clb("-l")
+      )
+
+  val wcCltOutput = CommandOutputParameter(
+    id = "wc-stdOut",
+    `type` = Some(Coproduct(CwlType.File)),
+    outputBinding = Some(CommandOutputBinding(glob = Some(Coproduct[Glob]("wc-stdOut.txt")))))
+
+  val wcClt =
+    CommandLineTool(
+      `class` = "CommandLineTool".narrow,
+      stdout = Some(Coproduct[StringOrExpression]("wc-stdOut.txt")),
+      inputs = Coproduct(Array(wcFileCommandInput)),
+      outputs = Coproduct(Array(wcCltOutput)),
+      arguments = Some(wcArgs),
+      requirements = inlineJScriptRequirements
     )
 
-    val m = new Workflow(
-      None,
-      `class` = "Workflow",
-      inputs = _inputs,
-      outputs = _outputs
-      steps = Array.empty[WorkflowStep].inject[WorkflowSteps])
+
+  val wcWorkflowInput = WorkflowStepInput(
+    id = "file",
+    source = Some(Coproduct("ps/ps-stdOut")))
+
+  val wcWorkflowStep = WorkflowStep(
+    id = Some("wc"),
+    in = Coproduct[WorkflowStep.Inputs](Array(wcWorkflowInput)),
+    out = Coproduct[WorkflowStep.Outputs](Array(WorkflowStepOutput("wc-stdOut"))),
+    run = Coproduct[WorkflowStep.Run](wcClt))
+
+
+  /**
+    * Workflow
     */
-  }
+  val outputCgrep =
+    WorkflowOutputParameter(
+      id = Some("cgrep-stdOut"),
+      `type` = Some(Coproduct[MyriadOutputType](CwlType.File)),
+      outputSource = Some(Coproduct("#cgrep/cgrep-stdOut"))
+    )
 
+  val outputWc =
+    WorkflowOutputParameter(
+      id = Some("wc-stdOut"),
+      `type` = Some(Coproduct[MyriadOutputType](CwlType.File)),
+      outputSource = Some(Coproduct("#wc/wc-stdOut"))
+    )
+
+  val _outputs = Coproduct[WorkflowOutput](Array(outputCgrep, outputWc))
+
+  val workflowPatternInput = InputParameter(id = Some("pattern"), `type` = Some(Coproduct[MyriadInputType](CwlType.String)))
+
+  val _inputs = Coproduct[WorkflowInput](Array(workflowPatternInput))
+
+  val threeStepWorkflow =
+    new Workflow(
+      Some(CwlVersion.Version1),
+      `class` = "Workflow".narrow,
+      inputs = _inputs,
+      outputs = _outputs,
+      steps = Coproduct[WorkflowSteps](Array(psWfStep, grepWfStep, wcWorkflowStep)))
```

Comment:
relative path that is resolved by [Schema Salad](http://www.commonwl.org/v1.0/SchemaSalad.html#Identifier_resolution) and passed in the input "ps/stdOut" `WorkFlowStepInput`'s source